test/utils/session-core.test.js

```diff
@@ -0,0 +1,249 @@
+import {
+  getSolidDataset,
+  createContainerAt,
+  mockSolidDatasetFrom,
+  mockThingFrom,
+  createSolidDataset,
+  setThing,
+  saveSolidDatasetInContainer
+} from '@inrupt/solid-client';
+import { expect, vi, it, describe, afterEach, beforeEach } from 'vitest';
+import { getContainerUrl } from '../../src/utils/network/session-helper';
+import { INTERACTION_TYPES } from '../../src/constants';
+
+vi.mock('@inrupt/solid-client', async () => {
+  const actual = await vi.importActual('@inrupt/solid-client');
+  return {
+    ...actual,
+    getSolidDataset: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    createContainerAt: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url))),
+    saveSolidDatasetInContainer: vi.fn((url) => Promise.resolve(mockSolidDatasetFrom(url)))
+  };
+});
+
+const getSolidDatasetRejected = vi.fn(() => Promise.reject(Error('dataset not found')));
```

Comment:
It's probably not the best idea to globally declare mocks like this. You'll get confusing performance when it gets cleared by `vi.clearAllMocks()`. You can just declare the mocks when you use them.

Reply:
Alright, I'll pull out the ones I'm using here outside of these global declarations.